### PR TITLE
FIX: Sort topic timer and bookmark time options

### PR DESCRIPTION
**Pull request by martin-brennan**
* remove 3 month option for topic timer
* move relative time input inside the  custom
  date and time shortcut
* make sure special options are always at the bottom

image

### CODE REVIEW

File: app/assets/javascripts/discourse/app/components/time-shortcut-picker.js [hunk 1]
```diff
@@ -197,34 +198,40 @@ export default Component.extend({
...
-    }
+    customOptions.forEach((opt) => {
+      if (!opt.timeFormatted && opt.time) {
+        opt.timeFormatted = opt.time.format(I18n.t(opt.timeFormatKey));
+      }
+    });
+
+    options = options.concat(customOptions);
+    options.sort((a, b) => {
+      return a.time < b.time ? -1 : a.time > b.time ? 1 : 0;
```
Comment (tgxworld) [inline app/assets/javascripts/discourse/app/components/time-shortcut-picker.js:209]: Can we avoid nesting two inlines if here? It is quite hard to follow.

File: config/locales/client.en.yml [hunk 2]
```diff
@@ -2460,13 +2461,13 @@ en:
...
-        two_weeks: "Two Weeks"
+        two_weeks: "Two weeks"
         next_month: "Next month"
-        two_months: "Two Months"
-        three_months: "Three Months"
-        four_months: "Four Months"
-        six_months: "Six Months"
-        one_year: "One Year"
+        two_months: "Two months"
+        three_months: "Three months"
```
Comment (tgxworld) [inline config/locales/client.en.yml:2467]: Are we still using this? I think we removed this option in the PR.

Comment (martin-brennan) [reply config/locales/client.en.yml:2467]: Yes it is still used by future date input selector component.

File: app/assets/javascripts/discourse/app/components/time-shortcut-picker.js (hunk 1 above)
Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/time-shortcut-picker.js]: @tgxworld fixed now

Comment (martin-brennan) [reply app/assets/javascripts/discourse/app/components/time-shortcut-picker.js]: I find the JS sorting thing annoying to read as well, but yeah the double ternery was too much

**Review by tgxworld — APPROVED** [review]
